tests/test_transforms/test_inference.py

```diff
@@ -0,0 +1,1664 @@
+                {"create": {"res_target"}},
+            ),
+            (
+                MaxAbsScalerTransform(in_column="target", mode="macro", inplace=True),
+                "regular_ts",
+                {"change": {"target"}},
+            ),
+            (
+                MinMaxScalerTransform(in_column="target", mode="macro", inplace=False, out_column="res"),
+                "regular_ts",
+                {"create": {"res_target"}},
+            ),
+            (
+                MinMaxScalerTransform(in_column="target", mode="macro", inplace=True),
+                "regular_ts",
+                {"change": {"target"}},
+            ),
+            (
+                RobustScalerTransform(in_column="target", mode="macro", inplace=False, out_column="res"),
+                "regular_ts",
+                {"create": {"res_target"}},
+            ),
+            (
+                RobustScalerTransform(in_column="target", mode="macro", inplace=True),
+                "regular_ts",
+                {"change": {"target"}},
+            ),
+            (
+                StandardScalerTransform(in_column="target", mode="macro", inplace=False, out_column="res"),
+                "regular_ts",
+                {"create": {"res_target"}},
+            ),
+            (
+                StandardScalerTransform(in_column="target", mode="macro", inplace=True),
+                "regular_ts",
+                {"change": {"target"}},
+            ),
+            (
+                YeoJohnsonTransform(in_column="target", mode="macro", inplace=False, out_column="res"),
+                "regular_ts",
+                {"create": {"res_target"}},
+            ),
+            (YeoJohnsonTransform(in_column="target", mode="macro", inplace=True), "regular_ts", {"change": {"target"}}),
+            # timestamp
+            (
+                DateFlagsTransform(out_column="res"),
+                "regular_ts",
+                {"create": {"res_day_number_in_week", "res_day_number_in_month", "res_is_weekend"}},
+            ),
+            (
+                FourierTransform(period=7, order=2, out_column="res"),
+                "regular_ts",
+                {"create": {"res_1", "res_2", "res_3", "res_4"}},
+            ),
+            (HolidayTransform(out_column="res"), "regular_ts", {"create": {"res"}}),
+            (
+                TimeFlagsTransform(out_column="res"),
+                "regular_ts",
+                {"create": {"res_minute_in_hour_number", "res_hour_number"}},
+            ),
+        ],
+    )
+    def test_transform_train_new_segments(self, transform, dataset_name, expected_changes, request):
+        ts = request.getfixturevalue(dataset_name)
+        self._test_transform_train_new_segments(
+            ts, transform, train_segments=["segment_1", "segment_2"], expected_changes=expected_changes
+        )
+
+    @to_be_fixed(raises=NotImplementedError, match="Per-segment transforms can't work on new segments")
```

Comment:
Why this test is separate from the others?

Reply:
Because it in `to_be_fixed` section. This can be done in theory, but we fail this test.
For instance, for decomposition transforms we can't do it in theory because we fit separate model for each segment.